### DISCUSSION TITLE
Firewall logs are not parsed after PFsense update

### DISCUSSION THREAD

**Discussion by sleepingbel** [Security-Onion-Solutions/securityonion#3009] [Unsupported Versions] [answered]
Hello all,

Security Onion standalone iso install 2.3.10 upgraded to 2.3.21
8 VCores
16 GB Vram
2 TB disk space

After a pfsense update to version 21.02, Security Onion has stopped parsing the firewall logs.

The logs from pfsense are still comming in, this i see in discover in the so-syslog index.

The log messages are like this now:
![new](https://user-images.githubusercontent.com/29328171/108345429-8b60fe00-71de-11eb-9d48-07b5ea9918d4.PNG)

I tried the following parsing but got an error:
^<%{INT:syslog.priority}>%{DATA:syslog.timestamp} %{WORD:source.application}\[%{DATA:pid}\]: %{GREEDYDATA:real_message}$

This is the error
{"error":{"root_cause":[{"type":"parse_exception","reason":"Failed to parse content to map"}],"type":"parse_exception","reason":"Failed to parse content to map","caused_by":{"type":"json_parse_exception","reason":"Unrecognized character escape '[' (code 91)\n at [Source: (org.elasticsearch.common.bytes.AbstractBytesReference$MarkSupportingStreamInputWrapper); line: 1, column: 755]"}},"status":400}
-bash: error:root_cause:[type:parse_exception]: command not found

The reason is the square brackets arround the pid, how can i solve this?

Regards
Bart

**Comment by sleepingbel** [comment, marked as answer]
Found the solution,

Ad this rule to the syslog parse file
`"^<%{INT:syslog.priority}>%{DATA:syslog.timestamp} %{WORD:source.application}\\[%{DATA:pid}\\]: %{GREEDYDATA:real_message}$",
`

And the firewall files are parsed

**Comment by dougburks** [comment]
Thanks @sleepingbel 

I've created the following issue:
https://github.com/Security-Onion-Solutions/securityonion/issues/3033

and I've committed your fix, but made the PID optional so that it will be backwards compatible with previous versions of pfsense:
https://github.com/Security-Onion-Solutions/securityonion/commit/88eb5b1d617ad1455ae39e015ade003d7d28ae41

This should be included in the upcoming 2.3.30 release.
